modules/wallet/update.go

```diff
@@ -18,10 +18,60 @@ type historicOutput struct {
 	val types.Currency
 }
 
+// isFutureWalletAddress is a helper function that checks if an UnlockHash is
+// derived from one of the wallet's future keys. If it is all future keys up to
+// this one will be added to the spendable keys or a blockhain rescan will be
+// initiated
+func (w *Wallet) isFutureWalletAddress(uh types.UnlockHash) bool {
+	if _, exists := w.futureKeys[uh]; !exists {
+		return false
+	}
+
+	// generate and add spendable keys until the future key is present in w.keys
+	foundKey := false
+	for i := 0; i < 1000; i++ {
+		_, err := w.nextPrimarySeedAddress(w.dbTx)
+		if err != nil {
+			return false
+		}
+
+		if _, exists := w.keys[uh]; exists {
+			foundKey = true
+			break
+		}
+	}
+
+	// if future key was not among the next 1000 keys rescan the blockchain
+	if !foundKey {
```

Comment:
@DavidVorick suggested that the lookahead is 5000 keys + 10% of the current seedProgress and if the key is not found in the first 1000 keys of the lookahead then the blockchain should be scanned again. That's why I included that if statement and also part of the reason for the `updateFutureKeys` function since the size of the lookahead is dynamic due to the 10%.

If that is not necessary I could fill up the futureKeys map to 5000 and only increase it if a new primary seed address is generated.